Move logs to dir

mikechristie opened the pull request:
The ceph selinux policy expects the logs to be in a dir under /var/log,
so this just moves our logs to /var/log/rbd-target-gw, so we do not have
to special case the ceph iscsi stuff in ceph-selinux.

mikechristie commented:
This PR will be used by this patch:

[Will do a pull for this patch when I know the dir patches are ok]
https://github.com/mikechristie/ceph/commit/7c3ec8105887e35f4d137868cddafbdf80066554

dillaman reviewed (approved):
lgtm

mikechristie commented:
Don't merge yet.

For the non-rpm install case, I need to add a fix so the /var/log/rbd-target-gw dir is made for the user.

mikechristie commented:
Updated patch uploaded. It will create the log dir when you do

python setup install

dillaman commented on setup.py:
```diff
@@ -42,6 +42,7 @@ def run(self):
     scripts=[
         "rbd-target-gw.py"
     ],
+    data_files=[("/var/log/rbd-target-gw", [])],
```
Should it be added to the spec file as well?

mikechristie replied:
Sorry. I missed this question.

We do not need to do a mkdir in the install section but we need a line in the files section:

%files
.....
%attr(3770,ceph,ceph) %dir /var/log/rbd-target-gw

and:

%attr(3770,ceph,ceph) %dir /var/log/rbd-target-api

in ceph-iscsi-cli

I am not sure what the permissions/user needs to be yet. I had just copied the values from the downstream ceph.spec when I was testing.

Checking now.

dillaman replied:
Yeah, I was just asking if the spec needed to create the directory (not to copy that exact line).

dillaman commented:
@mikechristie ping -- does this still require the spec change prior to merge?

Commit:
Move logs to dir

The ceph selinux policy expects the logs to be in a dir under
/var/log, so this just moves our logs to /var/log/rbd-target-gw,
so we do not have to special case the ceph iscsi stuff in ceph-selinux.

mikechristie commented:
Pushed updated PR with spec file change.
